### DISCUSSION TITLE
A question from a new user: how to use it?

### DISCUSSION THREAD

**Discussion by OnkelTem** [badges/shields#8185] [Q&A] [answered]
Hi!

I  need to add a simple npmjs badge for my project in the README.md, like this one: ![](https://camo.githubusercontent.com/d426a07bc8748f87302df1562db3dc7bb17f9a8d3a9d75436a3b7610938ee73a/68747470733a2f2f696d672e736869656c64732e696f2f6e706d2f762f77696e73746f6e2e7376673f7374796c653d666c61742d737175617265)

But after 30 minutes of seeking I'm about to beat my head against the wall to figure out how to use https://shields.io/

1) The search on https://shields.io/ doesn't find _any_ of my projects. So I cannot see which icons are available.
2) I don't see the "Npm" category in this list:
![image](https://user-images.githubusercontent.com/114060/178103224-701c6589-80f5-4d3c-b37d-0c8a294668fe.png)
So I can't understand how the authors of the above mentioned badge got it.
3) I cannot find also a way to ask https://shields.io/ to look up for my project and return the badge.

Could you please provide any clue?

**Comment by calebcartwright** [comment, marked as answer]
Shields is, in many ways, a provider of static content in the form of badges (typically svgs). These badges are served up at https://img.shields.io/route/to/badge/and/parameters, such as the static https://img.shields.io/badge/foo-bar-blue ![](https://img.shields.io/badge/foo-bar-blue)

The Shields.io website is simply a catalog or library of the types of badges that we provide. The website does not mention your specific projects, nor my projects, nor anyone else's projects, but it simply lists the _types_ of badges that are available, and the site also provides a tool that help users construct their corresponding badge url for _their_ projects/packages/etc..

We provide quite a lot of different types of badges, so they are grouped into categories. Please note that a category is different from a specific badge, for example, we have a build service category that contains badges for services related to builds/pipelines and you can view those by clicking the "Build" category link, or by navigating directly there: https://shields.io/category/build, and there's badges for many build/pipeline tools like AppVeyor, CircleCI, TravisCI, and more (these are the actual badges)

![image](https://user-images.githubusercontent.com/13042488/178110536-4d5bcacb-4c9f-49d4-960a-ee21ff775e82.png)

We also have a search box at the type of the page, where you can use keywords to try to find a particular type of badge. The types of things you should be searching for are terms like "github" or "npm" however, not the specific name of your GitHub repo or npm package as Shields.io does not have any insight into the full set of names of packages uploaded to the npm registry, every repository on GitHub, etc.

If you search for "npm" in the search box, you will see the available badges that are related to npm:

![image](https://user-images.githubusercontent.com/13042488/178110634-aed28bcc-0885-44e4-a665-1dd58ab2305b.png)

In either case (search or navigate), these pages show you the specific badges that are available, including the routes. You can click on any one of them to open up a dialog which we call the badge builder window, and the screenshot shows the badge builder window for the npm version badge which seems to be what you'd like to use:

![image](https://user-images.githubusercontent.com/13042488/178110698-b696c348-4746-4ec6-b774-91788924f852.png)


Here, you enter the relevant parameters, in this case the name of the desired package on the npm registry. I'll use one of our packages, `badge-maker` for an example. Once all required parameters have been provided, the window will show you a preview of your badge and you can use the "Copy Badge URL" button to get the corresponding url for the specific badge

![image](https://user-images.githubusercontent.com/13042488/178110758-d2a1d5e6-a0c9-4c36-9b05-d5b7d633ac70.png)


This produces the following:
https://img.shields.io/npm/v/badge-maker
![](https://img.shields.io/npm/v/badge-maker)

Which is the same pattern you would want to use, but you'd replace our package name `badge-maker` with the name of your package
